pkg/R/DataFrame.R

```diff
@@ -583,3 +581,29 @@ setMethod("foreachPartition",
             rdd <- toRDD(x)
             foreachPartition(rdd, func)
           })
+
+
+############################## DSL ##################################
+
+setMethod("$", signature(x = "DataFrame"),
+          function(x, name) {
+            column(callJMethod(x@sdf, "col", name))
+          })
+
+setGeneric("select", function(x, col, ...) { standardGeneric("select") } )
+
+setMethod("select", signature(x = "DataFrame", col = "character"),
+          function(x, col, ...) {
+            sdf <- callJMethod(x@sdf, "select", col, toSeq(...))
+            dataFrame(sdf)
+          })
+
+setMethod("select", signature(x = "DataFrame", col = "Column"),
+          function(x, col, ...) {
+            jcols <- lapply(list(col, ...), function(x) {
```

Comment:
Can we change the argument to the inner function to something other than `x`? The scope works out fine, but it's confusing to have `x` be a `DataFrame` outside of the function while also being a `Column` inside the function.

Even something like
```R
function(c) {
  c@jc
}
```